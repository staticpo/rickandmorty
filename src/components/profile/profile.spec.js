import React from 'react';
import { shallow, render, mount } from 'enzyme';
import escapeSnapshot from '../../../tests/commons';
import Profile from './index';

describe('Profile (Snapshot)', () => {
  const defaultProps = {
    id: 20,
    name: "Ants in my Eyes Johnson",
    status: "unknown",
    species: "Human",
    type: "Human with ants in his eyes",
    gender: "Male",
    origin: {
      name: "Origin",
      type: "type",
      dimension: "dimension",
    },
    location: {
      name: "Interdimensional Cable",
      type: "type",
      dimension: "dimension",
    },
    image: "https://rickandmortyapi.com/api/character/avatar/20.jpeg",
    episodes: [
      "Pilot"
    ],
  };

  const propsMissingData = {
    id: 20,
    name: "Ants in my Eyes Johnson",
    status: "unknown",
    species: "Human",
    type: "Human with ants in his eyes",
    gender: "Male",
    origin: {
      name: "unknown",
    },
    location: {
      name: "Interdimensional Cable",
    },
    image: "https://rickandmortyapi.com/api/character/avatar/20.jpeg",
    episodes: [
      "Pilot"
    ],
  };

  it('renders with all data', () => {
    const component = shallow(<Profile {...defaultProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('renders with some missing data', () => {
    defaultProps.currentPage = 10;
    defaultProps.prev = 9;
    const component = shallow(<Profile {...propsMissingData} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
