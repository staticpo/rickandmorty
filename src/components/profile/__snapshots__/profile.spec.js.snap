// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile (Snapshot) renders with all data 1`] = `
"<div className='info__frame'>
  <h1 className='info__title'>
    Ants in my Eyes Johnson
     (
    unknown
    )
  </h1>
  <img src='https://rickandmortyapi.com/api/character/avatar/20.jpeg' alt='Ants in my Eyes Johnson' />
  <p>
    Human
     - Human with ants in his eyes
  </p>
  <p>
    Gender: 
    Male
  </p>
  <details>
    <summary>
      Location
    </summary>
    <p>
      Interdimensional Cable
    </p>
    <p>
      Type: 
      type
    </p>
    <p>
      Dimension: 
      dimension
    </p>
  </details>
  <details>
    <summary>
      Origin
    </summary>
    <p>
      Origin
    </p>
    <p>
      Type: 
      type
    </p>
    <p>
      Dimension: 
      dimension
    </p>
  </details>
  <details>
    <summary>
      Episodes
    </summary>
    <p>
      Pilot
    </p>
  </details>
</div>"
`;

exports[`Profile (Snapshot) renders with some missing data 1`] = `
"<div className='info__frame'>
  <h1 className='info__title'>
    Ants in my Eyes Johnson
     (
    unknown
    )
  </h1>
  <img src='https://rickandmortyapi.com/api/character/avatar/20.jpeg' alt='Ants in my Eyes Johnson' />
  <p>
    Human
     - Human with ants in his eyes
  </p>
  <p>
    Gender: 
    Male
  </p>
  <details>
    <summary>
      Location
    </summary>
    <p>
      Interdimensional Cable
    </p>
  </details>
  <details>
    <summary>
      Origin
    </summary>
    <p>
      unknown
    </p>
  </details>
  <details>
    <summary>
      Episodes
    </summary>
    <p>
      Pilot
    </p>
  </details>
</div>"
`;
